rs/consensus/src/consensus/block_maker.rs

```diff
@@ -514,17 +516,50 @@ pub(crate) fn already_proposed(pool: &PoolReader<'_>, h: Height, this_node: Node
         .any(|p| p.signature.signer == this_node)
 }
 
+const DYNAMIC_DELAY_LOOK_BACK_DISTANCE: Height = Height::new(29);
+const DYNAMIC_DELAY_MAX_NON_RANK_0_BLOCKS: usize = 5;
+const DYNAMIC_DELAY_EXTRA_DURATION: Duration = Duration::from_secs(10);
+
+fn count_non_rank_0_blocks(pool: &PoolReader, block: Block) -> usize {
+    let max_height = block.height();
+    let min_height = if max_height > DYNAMIC_DELAY_LOOK_BACK_DISTANCE {
+        max_height - DYNAMIC_DELAY_LOOK_BACK_DISTANCE
+    } else {
+        Height::new(0)
+    };
+
+    pool.get_range(block, min_height, max_height)
```

Comment:
Doe this range only return blocks from the chain of the parent or any blocks with this height?